header-validator/data.source.js

```diff
@@ -1,262 +1,325 @@
+      "aggregation_keys": {"a": "3"}
+    }`,
+    expectedErrors: [{
+      path: ["aggregation_keys", "a"],
+      msg: "must be a hex128 (must match /^0[xX][0-9A-Fa-f]{1,32}$/)",
+    }],
+  },
+  // TODO: add tests for exceeding size limits
+
   {
-    source_event_id: '12340873456',
-    destination: 'https://example.com/foo',
+    name: "source-event-id-wrong-type",
+    json: `{
+      "destination": "https://a.test",
+      "source_event_id": 1
+    }`,
+    expectedErrors: [{
+      path: ["source_event_id"],
+      msg: "must be a string",
+    }],
+  },
+  {
+    name: "source-event-id-wrong-format",
+    json: `{
+      "destination": "https://a.test",
+      "source_event_id": "-1"
+    }`,
+    expectedErrors: [{
+      path: ["source_event_id"],
+      msg: "must be a uint64 (must match /^[0-9]+$/)",
+    }],
   },
-  // Destination contains a query string that will be ignored
+
   {
-    source_event_id: '12340873456',
-    destination: 'https://example.com?foo',
+    name: "debug-key-wrong-type",
+    json: `{
+      "destination": "https://a.test",
+      "debug_key": 1
+    }`,
+    expectedErrors: [{
+      path: ["debug_key"],
+      msg: "must be a string",
+    }],
+  },
+  {
+    name: "debug-key-wrong-format",
+    json: `{
+      "destination": "https://a.test",
+      "debug_key": "-1"
+    }`,
+    expectedErrors: [{
+      path: ["debug_key"],
+      msg: "must be a uint64 (must match /^[0-9]+$/)",
+    }],
   },
-  // Destination contains a fragment that will be ignored
+
   {
-    source_event_id: '12340873456',
-    destination: 'https://example.com#foo',
+    name: "priority-wrong-type",
+    json: `{
+      "destination": "https://a.test",
+      "priority": 1
+    }`,
+    expectedErrors: [{
+      path: ["priority"],
+      msg: "must be a string",
+    }],
+  },
+  {
+    name: "priority-wrong-format",
+    json: `{
+      "destination": "https://a.test",
+      "priority": "x"
+    }`,
+    expectedErrors: [{
+      path: ["priority"],
+      msg: "must be an int64 (must match /^-?[0-9]+$/)",
+    }],
   },
-  // Too many filters
+
   {
-    source_event_id: '0',
-    destination: 'https://example.com',
-    filter_data: (function () {
-      const obj = {}
-      for (let i = 0; i < 51; i++) {
-        obj[`${i}`] = []
-      }
-      return obj
-    })(),
+    name: "aggregatable-report-window-wrong-type",
+    json: `{
+      "destination": "https://a.test",
+      "aggregatable_report_window": 1
+    }`,
+    expectedErrors: [{
+      path: ["aggregatable_report_window"],
+      msg: "must be a string",
+    }],
+  },
+  {
+    name: "aggregatable-report-window-wrong-format",
+    json: `{
+      "destination": "https://a.test",
+      "aggregatable_report_window": "x"
+    }`,
+    expectedErrors: [{
+      path: ["aggregatable_report_window"],
+      msg: "must be an int64 (must match /^-?[0-9]+$/)",
+    }],
   },
-  // Too many filter values
+
   {
-    source_event_id: '0',
```

Comment:
Do we plan to add back the test cases for data length?

Reply:
Sure. I'll do that in a followup.